test/StripeResource.spec.js

```diff
@@ -32,4 +35,245 @@ describe('StripeResource', function() {
       expect(headers).to.not.include.keys('Stripe-Version');
     });
   });
+
+  describe('Retry Network Requests', function() {
+    // Use a real instance of stripe as we're mocking the http.request responses.
+    var realStripe = require('../lib/stripe')(utils.getUserStripeKey());
+    var options = {
+      host: stripe.getConstant('DEFAULT_HOST'),
+      path: '/v1/charges',
+      data: {
+        amount: 1000,
+        currency: 'usd',
+        source: 'tok_visa',
+        description: 'test'
+      },
+      params: 'amount=1000&currency=usd&source=tok_visa&description=test'
```

Comment:
why are we passing both data and params?

Reply:
Ah... This is fine, though might be nice to use `qs` to derive params from data...

Reply:
So in newer versions of nock (v10 and up) you can pass in an object which the library will automatically parse as a querystring. Unfortunately after downgrading to v9 (for Node v4 compatibility) it looks like I need to provide both a querystring and a data object here.